terminal: Evolution on input printing and flushing

doniacld opened the pull request:
- Add carriage return
- Add methods for writing and flushing input

I have been working with @gdey and it would be great if he can review it.

doniacld committed:
terminal: Evolution input printing and flushing

gdey reviewed (approved):
LGTM

There is a minor change though, we no longer need two function fro flushing the input to the serial and screen, so it would make sense to combine `writeInput` and  `flushInput`

doniacld commented:
> LGTM
> 
> There is a minor change though, we no longer need two function fro flushing the input to the serial and screen, so it would make sense to combine `writeInput` and `flushInput`

I just moved the write and flush in one method and added some comments. Let me know what you think after these las changes.

deadprogram commented on tutorial/terminal/main.go:
```diff
@@ -49,36 +62,54 @@ func main() {
 				input[i] = data
 
-				terminal.WriteByte(data)
+				_ = terminal.WriteByte(data)
```
Why all the `_ =`?

doniacld replied:
My linter was unhappy with uncaught errors 🤷‍♀️ I can revert them if you think it's more readable.

deadprogram replied:
it is very unreadable with all the `_` IMO :crying_cat_face:

doniacld replied:
Fixed! :)

doniacld commented:
Can I add the label `hacktoberfest-accepted` ? 🥺

deadprogram commented:
Thanks @doniacld now merging!
